packages/vkui/src/components/Progress/Progress.tsx

```diff
@@ -24,12 +25,20 @@ export interface ProgressProps extends HTMLAttributesWithRootRef<HTMLDivElement>
   /**
    * Стиль отображения прогрессбара
    */
-  appearance?: 'accent' | 'positive' | 'negative';
+  appearance?: 'accent' | 'positive' | 'negative' | 'custom';
+  /**
+   * Кастомный цвет прогрессбара. Используется только с `appearance="custom"`
+   */
+  color?: string;
```

Comment:
два свойства может быть не интуитивно, можно объединить

```suggestion
  appearance?: LiteralUnion<'accent' | 'positive' | 'negative' | `--${string}`, string>;
```

```tsx
<Progress appearance="#fff" />
```

```tsx
<Progress appearance="--my-variable" />
```

Reply:
Мне кажется объединение может в будущем быть неудобным. Вдруг понадобится красить фон прогрессбара в кастомный цвет, тогда уже одним свойством не отделаться. Чтобы уменьшить количество вопросов пользователей можно исключать color из типа, если `appearance` не `custom`. Тогда ts будет предупреждать заранее

Reply:
> Вдруг понадобится красить фон прогрессбара в кастомный цвет

Не  совсем понял кейс, у нас же `appearance` задаёт только цвет самой полоски прогресса. Если понадобится и подложку под ним красить, то это отдельное свойство будет, `backgroundColor`, например.

Добавить новое свойство всегда успеется, сейчас задачу можно решить через расширение значений

`appearance` и `color` два свойства об одном и том же, оба отвечают за цвет

> Чтобы уменьшить количество вопросов пользователей можно исключать color из типа, если appearance не custom.

Женерики мощная вещь, но в случае с Progress это усложнение, я бы избегал для простых задач прибегание к женерикам

Reply:
Понял, звучит логично, переделал